resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,376 @@
+    RussiaTimeZone11
+    NewZealandStandardTime
+    UTC12
+    FijiStandardTime
+    KamchatkaStandardTime
+    ChathamIslandsStandardTime
+    UTC13
+    TongaStandardTime
+    SamoaStandardTime
+    LineIslandsStandardTime
+
+}
+
+#endRegion Enums
+function Get-ValidTimeZone {
+    param (
+        [Parameter()]
+        # keep it string to avoid enum issues
+        [string] $TimeZone = ((Get-TimeZone).Id -replace '[\+\s\-\(\)\.]', ''),
+
+        # switch for Get() method
+        [Parameter()]
+        [switch] $NoValid
+    )
+
+    $list = (Get-TimeZone -ListAvailable).Id
+
+    $trimmedVersion = $list -replace '[\+\s\-\(\)\.]', ''
+
+    if ($trimmedVersion -contains $TimeZone) {
+        if ($NoValid.IsPresent) {
+            return $TimeZone
+        }
+
+        return $list[$trimmedVersion.IndexOf($TimeZone)]
+    } else {
+        throw 'Invalid time zone. Please provide a valid time zone without spaces and special characters.'
+    }
+}
+
+#region Classes
+<#
+.SYNOPSIS
+    DSC Resource to manage Windows Time settings.
+
+.DESCRIPTION
+    This `Time` DSC Resource allows you to manage the time zone, automatic time zone update, and system tray date/time visibility settings on a Windows machine.
+
+.PARAMETER TimeZone
+    The time zone to set on the machine. The value should be a valid time zone ID from the list of time zones (Get-TimeZone -ListAvailable).Id. The default value is the current time zone.
+
+.PARAMETER SetTimeZoneAutomatically
+    The method to use to set the time zone automatically. The value should be a boolean.
+
+.PARAMETER ShowSystemTrayDateTime
+    Whether to show the date and time in the system tray. The value should be a boolean. The default value is `$true`.
+
+.PARAMETER NotifyClockChange
+    Whether to notify the user when the time changes. The value should be a boolean.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -Name Time -Method Set -Property @{ TimeZone = "Pacific Standard Time"; SetTimeZoneAutomatically = "NTP"; ShowSystemTrayDateTime = $true }
+
+    This example sets the time zone to Pacific Standard Time, sets the time zone to be updated automatically using NTP, and shows the date and time in the system tray.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -Name Time -Method Get -Property {}
+
+    This example gets the current time settings on the machine.
+
+.EXAMPLE
+    PS C:\> Invoke-DscResource -Name Time -Method Test -Property @{ TimeZone = "Pacific Standard Time"}
+
+    This example tests whether the time zone is set to Pacific Standard Time.
+#>
+[DscResource()]
+class Time {
+    # TODO: Track issue 125 on PSDesiredStateConfiguration repository to add a ValidateSet for time zones
+    [DscProperty(Key)]
+    [TimeZoneTable] $TimeZone = ((Get-TimeZone).Id -replace '[\+\s\-\(\)\.]', '')
+
+    [DscProperty()]
+    [nullable[bool]] $SetTimeZoneAutomatically
+
+    [DscProperty()]
+    [nullable[bool]] $ShowSystemTrayDateTime
+
+    [DscProperty()]
+    [nullable[bool]] $NotifyClockChange
```

Comment:
I feel that these should all be separate DSC resources. I think a user shouldn't have to specify the timezone if they only want to Show the datetime in the system tray.

Reply:
Separated into own class `Clock`.